core/scheduler/chore.go

```diff
@@ -116,8 +116,26 @@ func (w *Worker) Execute(chore Chore) {
 
 	log.Debugf("%s: spinning up [main] goroutine to execute chore `do' function...", chore)
 	wait.Add(1)
+	task, err := w.db.GetTask(chore.TaskUUID)
+	if err != nil {
+		panic(fmt.Errorf("failed to retrieve task '%s' from database: %s", chore.TaskUUID, err))
+	}
 	go func() {
 		chore.Do(chore)
+
+		job, err := w.db.GetJob(task.JobUUID)
+		if err != nil {
+			panic(fmt.Errorf("failed to retrieve job '%s' from database: %s", task.JobUUID, err))
+		}
+		retries := job.Retries
+		log.Errorf("Retries: %d", chore)
+		if rc != 0 && retries > 0 {
```

Comment:
Are we making sure to store the retries # in the database in case the process exits prematurely? Eg. we should build failure handling into the retries. With this in mind perhaps it would be a better approach to modify the task running selection to account for # retries < retries. It's OK to do this as a feature enhancement as well as long as it is accounted for.

Reply:
The retries configured for the job by the user are stored in the database, but it does make sense to look into the "current retry". I'm thinking, retries itself is a failure-handle and looking at the current lightweight nature of SHIELD, would it be a far stretch? Also, it will increase the complexity of the state of tasks which again would need modifications in other operations- unfailed vs half-failed vs retry failed. Another thing which I think would be more important is scheduling retries at different time intervals. Generally if something fails now, it usually is going to fail in the next few minutes (saying after looking at the enormous number of fail-checks in SHIELD code). That leaves network or storage issue to be the major factor which would not be different in same moment.  Maybe having each retry happen after a certain time interval would make it more useful?

Reply:
Created a [JIRA ticket](https://starkandwayne.atlassian.net/browse/SHIELD-6)